### PR TITLE
PIM-11035: modify the command to fit with UCS

### PR DESCRIPTION
**Pull request by guillaumefer-akeneo**
**Description (for Contributor and Core Developer)**

Modify the UpdateAuditData command to handle UCS and show data on the Data Flow Inbound graph

### CODE REVIEW

File: src/Akeneo/Connectivity/Connection/back/Infrastructure/Audit/UpdateAuditData.php [hunk 1]
```diff
@@ -83,4 +55,10 @@ private function updateProductEventCount(HourlyInterval $hourlyInterval): void
             new UpdateDataSourceProductEventCountCommand($hourlyInterval)
         );
     }
+
+    private function getDefaultStartDatetime(int $days)
```
Comment (RiffVentura) [inline src/Akeneo/Connectivity/Connection/back/Infrastructure/Audit/UpdateAuditData.php:59]: missing return type

File: src/Akeneo/Connectivity/Connection/back/Infrastructure/Audit/UpdateAuditData.php [hunk 2]
```diff
@@ -30,50 +28,24 @@ public function execute(): void
...
 
+        foreach ($hourlyIntervals as $hourlyInterval) {
             $this->updateProductEventCount($hourlyInterval);
         }
 
         $this->logger->info('End audit data purge');
     }
 
-    private function purgeOlderThanXDays(int $days): void
+    private function purgeOlderThanXDays(\DateTimeImmutable $before): void
```
Comment (RiffVentura) [inline src/Akeneo/Connectivity/Connection/back/Infrastructure/Audit/UpdateAuditData.php:47]:
```suggestion
    private function purgeEventsOlderThan(\DateTimeImmutable $before): void
```

File: src/Akeneo/Connectivity/Connection/back/Infrastructure/Symfony/Resources/config/Audit/persistence.yml [hunk 3]
```diff
@@ -35,3 +35,7 @@ services:
     Akeneo\Connectivity\Connection\Infrastructure\Audit\Persistence\DbalUpsertEventCountQuery:
         arguments:
             - '@database_connection'
+
+    Akeneo\Connectivity\Connection\Infrastructure\Audit\Persistence\DbalSelectLastConnectionsEventDatetimeQuery:
```
Comment (RiffVentura) [inline src/Akeneo/Connectivity/Connection/back/Infrastructure/Symfony/Resources/config/Audit/persistence.yml:39]: Should be deleted no?

File: src/Akeneo/Connectivity/Connection/back/Domain/ValueObject/HourlyInterval.php [hunk 4]
```diff
@@ -63,4 +63,29 @@ public function equals(self $hourlyInterval): bool
...
+
+    /**
+     * Returns an array of HourlyInterval instances representing hourly intervals between the start and end dates.
+     *
+     * @param \DateTimeInterface $startDateTime
+     * @param \DateTimeInterface $endDateTime
+     *
+     * @return HourlyInterval[]
+     */
+    public static function getHourlyIntervals(\DateTimeInterface $startDateTime, \DateTimeInterface $endDateTime): array
```
Comment (RiffVentura) [inline src/Akeneo/Connectivity/Connection/back/Domain/ValueObject/HourlyInterval.php:75]: This method should not be here IMO: this method creates an array of itself

I think you should put it directly into UpdateAuditData it is only needed there

File: src/Akeneo/Connectivity/Connection/back/tests/EndToEnd/Audit/Command/UpdateAuditDataCommandEndToEnd.php [hunk 5]
```diff
@@ -139,14 +139,49 @@ public function test_updates_audit_data_only_for_data_source_connection(): void
         Assert::assertEquals(0, $this->getAuditCount($destinationConnection->code(), EventTypes::PRODUCT_UPDATED));
     }
 
+    public function test_audit_data_creation_and_updates_for_10_days(): void
```
Comment (RiffVentura) [inline src/Akeneo/Connectivity/Connection/back/tests/EndToEnd/Audit/Command/UpdateAuditDataCommandEndToEnd.php:142]: Data provider would suit better here and avoid doing multiple tests inside one test

**Review by RiffVentura — COMMENTED** [review]
Missing CHANGELOG update

**Review by eoliveauakeneo — APPROVED** [review]

**Review by RiffVentura — APPROVED** [review]
CHANGELOG to update :smile:

**Pull request merged**